Port HTTPHeaderNames to the new serialization format

gavin-apple opened the pull request:
#### 3ecba4327009a8d93e79778babad98c85328bc67
<pre>
Port HTTPHeaderNames to the new serialization format
<a href="https://bugs.webkit.org/show_bug.cgi?id=252984">https://bugs.webkit.org/show_bug.cgi?id=252984</a>
rdar://105970730

Reviewed by NOBODY (OOPS!).

Ports the HTTPHeaderNames enum to the new serialization format.

* Source/WebCore/platform/network/HTTPParsers.h:
* Source/WebCore/platform/network/create-http-header-name-table:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Scripts/generate-serializers.py:
(parse_serialized_types):
(main):
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3ecba4327009a8d93e79778babad98c85328bc67

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/111719 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/20856 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/334 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/3510 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/120448 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/115780 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/22211 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/11935 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/86/builds/3259 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/117485 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/16492 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/99647 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/104681 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/115148 "Passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/98444 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/88/builds/223 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/45437 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/13332 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/89/builds/224 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/92650 "run-api-tests-without-change (failure)") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/13824 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/9661 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/19269 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/52222 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/15814 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebKit/DerivedSources-input.xcfilelist:
```diff
@@ -1,4 +1,5 @@
 # This file is generated by the generate-xcfilelists script.
+$(BUILT_PRODUCTS_DIR)/DerivedSources/WebCore/HTTPHeaderNames.serialization.in
```
I think this needs to use WEBCORE_PRIVATE_HEADERS_DIR to not break the internal build.

achristensen07 commented on Source/WebKit/DerivedSources.make:
```diff
@@ -456,6 +456,7 @@ $(WEB_PREFERENCES_PATTERNS) : $(WTF_BUILD_SCRIPTS_DIR)/GeneratePreferences.rb $(
 	$(RUBY) $< --frontend WebKit $(addprefix --template , $(WEB_PREFERENCES_TEMPLATES)) $(WEB_PREFERENCES)
 
 SERIALIZATION_DESCRIPTION_FILES = \
+	$(BUILT_PRODUCTS_DIR)/DerivedSources/WebCore/HTTPHeaderNames.serialization.in \
```
ditto

achristensen07 commented on Source/WebCore/WebCore.xcodeproj/project.pbxproj:
```diff
@@ -3,7 +3,7 @@
 	classes = {
 	};
-	objectVersion = 54;
+	objectVersion = 51;
```
Let's not change this in this PR.

achristensen07 commented on Source/WebKit/DerivedSources-output.xcfilelist:
```diff
@@ -1,4 +1,6 @@
 # This file is generated by the generate-xcfilelists script.
+$(BUILT_PRODUCTS_DIR)/DerivedSources/WebKit/(BUILT_PRODUCTS_DIR)/DerivedSources/WebKit/HTTPHeaderNames.serialization.in
+$(BUILT_PRODUCTS_DIR)/DerivedSources/WebKit//Users/gavin/code/wk_dev3/OpenSource/WebKitBuild/Debug/DerivedSources/WebKit/HTTPHeaderNames.serialization.in
```
These are not correct.  It should especially not contain "gavin"

achristensen07 replied:
I think just remove these. It 's correct below.

achristensen07 commented on Source/WebKit/DerivedSources.make:
```diff
@@ -455,7 +455,12 @@ all : $(WEB_PREFERENCES_FILES)
 $(WEB_PREFERENCES_PATTERNS) : $(WTF_BUILD_SCRIPTS_DIR)/GeneratePreferences.rb $(WEB_PREFERENCES_TEMPLATES) $(WEB_PREFERENCES)
 	$(RUBY) $< --frontend WebKit $(addprefix --template , $(WEB_PREFERENCES_TEMPLATES)) $(WEB_PREFERENCES)
 
+all : $(BUILT_PRODUCTS_DIR)/DerivedSources/WebKit/HTTPHeaderNames.serialization.in
```
I think you need to remove $(BUILT_PRODUCTS_DIR)/DerivedSources/WebKit/

achristensen07 commented on Source/WebKit/DerivedSources.make:
```diff
@@ -455,7 +455,12 @@ all : $(WEB_PREFERENCES_FILES)
 	$(RUBY) $< --frontend WebKit $(addprefix --template , $(WEB_PREFERENCES_TEMPLATES)) $(WEB_PREFERENCES)
 
+all : $(BUILT_PRODUCTS_DIR)/DerivedSources/WebKit/HTTPHeaderNames.serialization.in
+$(BUILT_PRODUCTS_DIR)/DerivedSources/WebKit/HTTPHeaderNames.serialization.in : $(BUILT_PRODUCTS_DIR)/WebCore.framework/PrivateHeaders/HTTPHeaderNames.in $(WebKit2)/Scripts/create-http-header-serialization-interface.py
```
I don't think $(BUILT_PRODUCTS_DIR)/WebCore.framework/PrivateHeaders/ is the correct place in the internal build.  use $(WebCorePrivateHeaders)

gavin-apple committed:
Port HTTPHeaderNames to the new serialization format
https://bugs.webkit.org/show_bug.cgi?id=252984
rdar://105970730

Reviewed by NOBODY (OOPS!).

Ports the HTTPHeaderNames enum to the new serialization format.

* Source/WebCore/platform/network/HTTPParsers.h:
* Source/WebCore/platform/network/create-http-header-name-table:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Scripts/generate-serializers.py:
(parse_serialized_types):
(main):
